include/tc/core/cuda/cuda_tc_executor.h

```diff
@@ -0,0 +1,118 @@
+#include "tc/core/utils/dlpack.h"
+#include "tc/lang/parser.h"
+
+#include <dlpack/dlpack.h>
+
+namespace tc {
+
+using PpcgppCompilationCommand = std::function<std::string(
+    std::unique_ptr<tc::polyhedral::Scop>&& scop,
+    const std::vector<const DLTensor*>& inputs,
+    const MappingOptions& options)>;
+
+class CudaTcExecutor : public ::tc::TcExecutor {
+ public:
+  CudaTcExecutor(
+      const std::string& def,
+      const std::vector<const DLTensor*>& inputsInfo)
+      : TcExecutor(def, inputsInfo) {}
+  CudaTcExecutor(
+      lang::TreeRef tree,
+      const std::vector<const DLTensor*>& inputsInfo)
+      : TcExecutor(tree, inputsInfo) {}
+  ~CudaTcExecutor() {}
+
+  CudaTcExecutor(CudaTcExecutor&&) = delete;
+  CudaTcExecutor& operator=(CudaTcExecutor&&) = delete;
+  CudaTcExecutor(const CudaTcExecutor&) = delete;
+  CudaTcExecutor& operator=(const CudaTcExecutor&) = delete;
+
+  // Can only be called once with specific kernel options.  Input sizes are
+  // set up as constructor argument and output sizes are inferred.
+  //
+  // If you need another kernel for another Tc or another inputs, outputs,
+  // options then just instantiate another CudaTcExecutor.
+  // This is because for the time being we fully specialize all the sizes and
+  // strides at runtime.
+  // @{
+  void compile(const std::string& options) override {
+    compile(MappingOptions(options));
+  }
+  void compile(const tc::MappingOptions& options);
+  // @}
+
+  // Run can be called multiple times given a compilation, inputs are allowed
+  // to change in that their data pointer is allowed to change.
+  // Sizes and strides must remain constant otherwise this is an error
+  // The only thing that is allowed to change across runs is the input
+  // and output pointers base address.
+  // It is the caller's responsibility to ensure proper non-aliasing (or
+  // advanced aliasing) properties of the input and output tensors.
+  // if profile is set the kernel runtime (nanoseconds) is returned
+  Duration run(
+      const std::vector<const DLTensor*>& inputs,
+      const std::vector<DLTensor*>& outputs,
+      bool profile = false) const;
+
+  // This is the "low-latency" mode in which we just propagate raw pointers to
+  // data in GPU address space.
+  // No tensor-related information can be checked so it is the user's
+  // responsibility to ensure that shapes and strides match. If the user
+  // doesn't then segfault will likely occur.
+  void uncheckedRun(
+      const std::vector<const void*>& inputs,
+      const std::vector<void*>& outputs) const;
+
+  bool hasRTCFun() {
+    return rtcFun.get() != nullptr;
+  }
+
+  // It is necessary to clear the RTC manually because it can throw and we
+  // can't have that in the destructor.
+  void clearRTC() {
+    if (!hasRTCFun()) {
+      return;
+    }
+    rtcFun->clear();
+  }
+
+  std::string kernelName() const {
+    return execInfo_.kernelName;
+  }
+
+ private:
+  void compileWithPetPpcg();
```

Comment:
why does this exist? didn't we remove petppcg? this header is looking a lot like back in October

Reply:
I don't know why this was not removed.  I removed it in a separate commit later.